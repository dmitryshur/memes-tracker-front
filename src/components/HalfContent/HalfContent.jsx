import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './HalfContent.scss';

const HalfContent = props => {
  const contentClick = () => {
    console.log('testing');
  };

  return (
    <div
      className={`half-content half-content-${props.side}`}
      onClick={contentClick}
      onKeyPress={contentClick}
      role="button"
      tabIndex="0"
    >
      <div className="half-content-logo" />
      <div className="half-content-title">
        {props.side === 'left' ? 'newest' : 'liked'}
      </div>
      <hr />
      <div className="half-content-subtitle">
        {props.side === 'left' ? 'Most recent memes' : 'Most liked memes'}
      </div>
    </div>
  );
};

HalfContent.propTypes = {
  side: PropTypes.string,
};

HalfContent.defaultProps = {};

export default HalfContent;
