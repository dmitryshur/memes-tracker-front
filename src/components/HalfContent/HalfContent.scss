$newest-color: #fff;
$liked-color: #47a3da;

.half-content {
  align-items: center;
  display: flex;
  flex-direction: column;
  left: 50%;
  outline: none;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  transition-duration: 1s;
  transition-property: all;
  z-index: 21;

  &-logo {
    background-position: center center;
    background-size: cover;
    height: 100px;
    margin-bottom: 10px;
    width: 100px;
  }

  &-title {
    font-size: 2.4rem;
    font-weight: 700;
    text-transform: uppercase;
  }

  hr {
    border: 0;
    height: 2px;
    margin: 6px 0;
    width: 100%;
  }

  &-newest {
    color: $newest-color;

    .half-content-logo {
      background-image: url('/images/newest-logo.png');
    }

    hr {
      background: #fff;
      background-image: linear-gradient(to right, #eee, #fff, #eee);
    }

    &.opened {
      left: 20%;
      transform: translate(-50%, -50%) scale(0.8);
    }
  }

  &-liked {
    color: $liked-color;

    .half-content-logo {
      background-image: url('/images/liked-logo.png');
    }

    hr {
      background: #47a3da;
      background-image: linear-gradient(to right, #45a0ca, #47a3da, #45a0ca);
    }

    &.opened {
      left: 80%;
      transform: translate(-50%, -50%) scale(0.8);
    }
  }

  &-subtitle {
    font-size: 1.8rem;
    white-space: nowrap;
  }

  &:hover {
    cursor: pointer;
  }
}
