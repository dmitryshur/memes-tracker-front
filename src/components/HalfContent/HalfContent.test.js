import React from "react";
import { shallow } from "enzyme";
import { HalfContent } from "./HalfContent.jsx";

test("testing HalfContent newest component", () => {
  const wrapper = shallow(<HalfContent section="newest" />);
  expect(wrapper.find(".half-content-newest").length).toBe(1);
  expect(wrapper.find('[data-id="newest"]').length).toBe(1);
  expect(wrapper.find(".half-content-title").text()).toBe("newest");
  expect(wrapper.find(".half-content-subtitle").text()).toBe("Most recent memes");
  expect(wrapper).toMatchSnapshot();
});

test("testing HalfContent newest component click simulation", () => {
  const setChosenSection = jest.fn();
  const wrapper = shallow(<HalfContent section="newest" setChosenSection={setChosenSection} />);

  wrapper.find(".half-content").simulate("click", {
    currentTarget: { dataset: { id: "newest" } }
  });
  expect(setChosenSection).toHaveBeenLastCalledWith("newest");
  expect(wrapper).toMatchSnapshot();
});

test("testing HalfContent liked component", () => {
  const wrapper = shallow(<HalfContent section="liked" />);
  expect(wrapper.find(".half-content-liked").length).toBe(1);
  expect(wrapper.find('[data-id="liked"]').length).toBe(1);
  expect(wrapper.find(".half-content-title").text()).toBe("liked");
  expect(wrapper.find(".half-content-subtitle").text()).toBe("Most liked memes");
  expect(wrapper).toMatchSnapshot();
});

test("testing HalfContent liked component click simulation", () => {
  const setChosenSection = jest.fn();
  const wrapper = shallow(<HalfContent section="liked" setChosenSection={setChosenSection} />);

  wrapper.find(".half-content").simulate("click", {
    currentTarget: { dataset: { id: "liked" } }
  });
  expect(setChosenSection).toHaveBeenLastCalledWith("liked");
  expect(wrapper).toMatchSnapshot();
});

test("testing HalfContent newest component when newest section is closed", () => {
  const wrapper = shallow(<HalfContent section="newest" openState="closed" />);
  expect(wrapper.find(".closed").length).toBe(1);
});

test("testing HalfContent newest component when newest section is opened", () => {
  const wrapper = shallow(<HalfContent section="newest" openState="opened" />);
  expect(wrapper.find(".opened").length).toBe(1);
});

test("testing HalfContent liked component when newest section is closed", () => {
  const wrapper = shallow(<HalfContent section="liked" openState="closed" />);
  expect(wrapper.find(".closed").length).toBe(1);
});

test("testing HalfContent liked component when newest section is opened", () => {
  const wrapper = shallow(<HalfContent section="liked" openState="opened" />);
  expect(wrapper.find(".opened").length).toBe(1);
});
