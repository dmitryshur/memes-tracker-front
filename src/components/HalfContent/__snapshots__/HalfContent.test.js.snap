// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing HalfContent liked component 1`] = `
<div
  className="half-content half-content-liked closed"
  data-id="liked"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="half-content-logo"
  />
  <div
    className="half-content-title"
  >
    liked
  </div>
  <hr />
  <div
    className="half-content-subtitle"
  >
    Most liked memes
  </div>
</div>
`;

exports[`testing HalfContent liked component click simulation 1`] = `
<div
  className="half-content half-content-liked closed"
  data-id="liked"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="half-content-logo"
  />
  <div
    className="half-content-title"
  >
    liked
  </div>
  <hr />
  <div
    className="half-content-subtitle"
  >
    Most liked memes
  </div>
</div>
`;

exports[`testing HalfContent newest component 1`] = `
<div
  className="half-content half-content-newest closed"
  data-id="newest"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="half-content-logo"
  />
  <div
    className="half-content-title"
  >
    newest
  </div>
  <hr />
  <div
    className="half-content-subtitle"
  >
    Most recent memes
  </div>
</div>
`;

exports[`testing HalfContent newest component click simulation 1`] = `
<div
  className="half-content half-content-newest closed"
  data-id="newest"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="half-content-logo"
  />
  <div
    className="half-content-title"
  >
    newest
  </div>
  <hr />
  <div
    className="half-content-subtitle"
  >
    Most recent memes
  </div>
</div>
`;
